R/teal_data_module-within.R

```diff
@@ -11,15 +13,10 @@
 #' `within` returns a `teal_data_module` object with a delayed evaluation of `expr` when the module is run.
 #'
 #' @examples
-#' tdm <- teal_data_module(
-#'   ui = function(id) div(id = shiny::NS(id)("div_id")),
-#'   server = function(id) {
-#'     shiny::moduleServer(id, function(input, output, session) {
-#'       shiny::reactive(teal_data(IRIS = iris))
-#'     })
-#'   }
-#' )
-#' within(tdm, IRIS <- subset(IRIS, Species == "virginica"))
+#' within(tdm, dataset1 <- subset(dataset1, Species == "virginica"))
+#'
+#' # in an app, use input-derived value in the expression
+#' within(tdm, dataset1 <- subset(dataset1, Species == species), species = input$species)
```

Comment:
@chlebowa this within call fails the check [here](https://github.com/insightsengineering/teal/actions/runs/7581248835/job/20648495135) . Should we move it to the comments section? WDYT?
```suggestion
#' # within(tdm, dataset1 <- subset(dataset1, Species == species), species = input$species)
```

Reply:
This example doesn't make much sense as the `teal_data_module` shouldn't be used "inside the app".

`tdm` itself is the app and the `within` calls on it should be performed outside any shiny environment and before the app is initialized.

They will then be applied when the `teal` app initializes and the `tdm` app is executed.

In the connectors we use it to "normalize" the resulting `teal_data` object

note that suggestion implies the line before is also removed.

```suggestion
#' # in an app, use variables defined outside the `teal_data_module` scope in the expression
#' valid_species = c("setosa", "versicolor")
#' within(tdm, dataset1 <- subset(dataset1, Species %in% species), species = valid_species)
```

Reply:
Sorry, I should have tested it.
Can anyone suggest a different example for `within` that utilizes `...`?

Reply:
what about the example @averissimo  suggested above ?

Reply:
This has been integrated. Resolving